pkg/codegen/schema/schema.go

```diff
@@ -567,6 +567,22 @@ func (fun *Function) NeedsOutputVersion() bool {
 	return fun.ReturnType != nil
 }
 
+// BaseProvider
+type BaseProvider struct {
+	// Name is the name of the provider.
+	Name string
+	// Version is the version of the provider.
+	Version semver.Version
+	// PluginDownloadURL is the URL to use to acquire the provider plugin binary, if any.
+	PluginDownloadURL string
```

Comment:
This one is different from the rest, since it doesn't make sense for a parameterized schema to have it's own plugin download url (since it shares a binary with the base provider). 

This implies that all base provider information will be duplicated into `BaseProvider`, not just information in slots needed by the parametrization.

Reply:
Yeh this is all the base provider information being duplicated. That's so we have enough information given _just_ the parameterised schema to lookup the base provider plugin.

Reply:
Makes sense. Thanks for clarifying.